src/reformat_libyuv.c

```diff
@@ -499,6 +504,16 @@ avifResult avifRGBImageUnpremultiplyAlphaLibYUV(avifRGBImage * rgb)
     return AVIF_RESULT_NOT_IMPLEMENTED;
 }
 
+avifResult avifRGBImageToF16LibYUV(avifRGBImage * rgb)
+{
+    const float scale = 1.0f / ((1 << rgb->depth) - 1);
+    HalfFloatPlane((const uint16_t*)rgb->pixels, rgb->rowBytes,
```

Comment:
I found that HalfFloatPlane() returns 0 on success, -1 on failure. So in theory we should check its return value. This means avifRGBImageToF16LibYUV() will also return `AVIF_RESULT_INVALID_ARGUMENT`, we will need to change avifRGBImageToF16() back to return avifResult.

However, in the current implementation of HalfFloatPlane(), it only returns -1 here:

```
  if (!src_y || !dst_y || width <= 0 || height == 0) {
    return -1;
  }
```

So we can verify we don't pass such invalid arguments to HalfFloatPlane(), and therefore this HalfFloatPlane() call cannot fail.

What do you think?

Reply:
Ah nice catch. I didn't notice that HalfFloatPlane() returned a return value (most libyuv functions don't).

I have updated the code to check the return value.